### DISCUSSION TITLE
현재 조직 내의 스타일 가이드 정리 및 개발 워크플로우 정리

### DISCUSSION THREAD

**Discussion by peanudge** [InnerCircleA/design-system#15] [Show and tell]
4주차 미션 관련해서 공유하면 좋을 것 같아서 discussion 따로 만들어 두겠습니다!

공유 가능하신 분들은 한번 공유해주시면 많은 도움이 될 것 같습니다. @Jeewhan @ppamppamman 

• 조직 내 프로젝트에 기반한 디자인 가이드 정리
• 디자인 가이드 토대로 컴포넌트 라이브러리 정리
• 개발 워크플로우 정리

**Comment by peanudge** [comment]
# 조직 내 스타일 가이드 정리와  컴포넌트 라이브러리 정리

사실 현재 제가 재직중인 회사에서는 새로운 디자인 시스템을 만들고 있지만 아직 오픈되지 않아서 참조하지 못했었습니다.
그래서 잘 정리되지 않은 과거 버젼의 figma 내용들을 참조하고 암묵적으로 따르고 있는 디자인 규칙들을 가시화하는데 초점을 두고 간단히 **버튼 사용 가이드라인**을 만드는 것을 목표로 미션을 진행해보았습니다.

https://github.com/InnerCircleA/design-system/tree/main/design-guide/peanut-lover

위 경로에 storybook을 이용해서 `.mdx` 기반 문서화와 컴포넌트 구현을 정리해보았습니다.

간단히 위 과정을 진행하면서 얻은 구체적인 현재 디자인 시안의 문제들은

- 현재 서비스는 앱서비스만 주력으로 개발하고 있기에 웹으로 확장하려면 더 풍부한 파운데이션 요소들이 필요할 것으로 보입니다.
- 정적인 디자인만 있지만 실제 개발에는 동적인 디자인 요소들도 필요한데 그러한 정보가 부족하다. (ex, active, hover, focus) 
- UI는 거의 유사하지만 쓰임이 달라 다른 컴포넌트로 분류되는 것들도 있어서 혼란을 가져오는 요소들이 있습니다.  

정리하자면, 이미 디자인된 시안들을 모아서 철학을 단기간에 만들기는 어려운 점이 많았고 디자인 시안들이 크게 변경 되었을 때 한번에 문서로 반영하려하니 큰 작업이 될 것처럼 느껴졌습니다. 

아마 디자이너 - 개발자 사이의 워크 플로우에 이런 문서화 불일치도 디자인시스템을 저하시키는 원인 될 수 있을 것 같습니다.

**Comment by ppamppamman** [comment]
## 조직 내 프로젝트에 기반한 스타일 가이드 정리
  조직의 스타일 가이드는 다음의 툴로 다뤄지고 있습니다.
  - figma : 디자이너의 스타일 정의 도구 + 정의한 스타일에 대한 값을 다루기 위한 도구
  - zeroheight : 디자인 시스템에 대한 도큐먼트
  - notion : 과거의 도큐먼트 + 커뮤니케이션 툴

  스타일 가이드에는 아래의 항목들이 정리되어 있습니다.
  - Layout : Grid system과 적용 기준, 메인 레이아웃, 사례 (반응형 레이아웃의 변화 사례)
  - Space : 유닛 사이즈와 기준, base size, 사용 사례
  - Typo : Font family, 프로퍼티와 단위 (size, line height, letter spacing), 사례 (type scales, styles)
  - Color: 팔레트, Semantic colors (버튼 인터렉션, 상태에 대한 컬러값, 텍스트 컬러값), 적용 방식와 예외 사례
  - Icons : svg viewbox가 통일된 형태로 정립
  
  크게 정리하면, 가이드 정립 기준, 가이드 사용 및 확장 방법, 사용 사례 정도로 정리해볼 수 있습니다.
  
현재 저희 조직의 스타일 가이드의 관리체계에서 발생하는 문제는 아래와 같습니다.
실제 작업이 진행되는 figma와 문서화를 담당하는 zeroheight의 관리가 각각 따로 이뤄집니다. 따라서 업데이트가 필요한 경우 따로따로 작업을 해줘야 하는 문제가 있고, 이에 따른 휴먼에러, 높은 커뮤니케이션 코스트가 발생할 여지가 있습니다. 레거시의 영향 / 리소스 부족의 영향으로 일부 정보가 Notion에서 파편화되어 존재하고, 시간이 지남에 따라 관리가 어려워지고 있습니다.


## 스타일 가이드 토대로 컴포넌트 라이브러리 정리
 
>  컴포넌트의 정의체계를 다루기에 시간이 약간 부족한 상황이라 우선 개발 단계에서의 문제점을 우선 간략하게 작성합니다. 시간이 되는대로 정의체계와 현재의 개발상황과 맞지 않는 지점을 추가로 작성할 생각입니다.  

  컴포넌트 라이브러리의 경우 ant.design을 토대로 빌드가 되고 있는 상황이라 대부분의 가이드는 antd를 존중하며 작성되고 있습니다. 조직에서 정의한 파운데이션 영역을 적용할 수 있는 부분에 대해서는 엄밀하게 적용하고 있습니다. 다만 문제가 되는 부분은 antd의 컴포넌트에서 적용을 지원하지 않아 실제 DOM으로부터 도출되는 클래스네임을 찾아 오버라이딩하는 부분이 있습니다. 또한 라이브러리의 버전 업데이트가 되며 스타일이 일부 변경되는 경우 트래킹이 어렵다는 문제가 있었습니다.

**Reply by peanudge** [reply]
>실제 작업이 진행되는 figma와 문서화를 담당하는 zeroheight의 관리가 각각 따로 이뤄집니다. 따라서 업데이트가 필요한 경우 따로따로 작업을 해줘야 하는 문제가 있고, 이에 따른 휴먼에러, 높은 커뮤니케이션 코스트가 발생할 여지가 있습니다. 레거시의 영향 / 리소스 부족의 영향으로 일부 정보가 Notion에서 파편화되어 존재하고, 시간이 지남에 따라 관리가 어려워지고 있습니다.

역시 문서화는 어디에서나 문제가 될 수 있군요. 저는 이 부분이 단순한 팀원들의 성실함의 문제일지 아니면 기술적 문제로 바라봐야할지 의문인체 항상 남아있는 것 같습니다. 그리고 궁금한 점이 생겼는데 이런 **디자인 가이드 관리는 디자이너들이 오너쉽**이 있을까요? 그리고 디자인 시스템을 관리하는 개발자들이 따로 있으려나요?
